LearnositySDK/Request/Init.cs

```diff
@@ -403,13 +403,33 @@ private void setServiceOptions()
 
                         JsonObject questionsApiActivity = new JsonObject();
 
-                        string signature = this.hashValue( new string[] {
-                            this.securityPacket.getString("consumer_key"),
-                            domain,
-                            this.securityPacket.getString("timestamp"),
-                            this.securityPacket.getString("user_id"),
-                            this.secret
-                        } );
+                        string[] arrayToHash;
+
+                        if (Tools.array_key_exists("expires", this.securityPacket))
+                        {
+                            arrayToHash = new string[] {
```

Comment:
@gonzalozawa in here, instead of writing the same code twice you could use generic list (https://msdn.microsoft.com/en-us/library/6sh2ey19.aspx) of strings and run `ToArray()` at the end of processing (https://msdn.microsoft.com/en-us/library/x303t819(v=vs.110).aspx).